internal/services/network/network_interface_resource.go

```diff
@@ -386,14 +398,27 @@ func resourceNetworkInterfaceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		payload.Properties.IPConfigurations = ipConfigs
 	}
 
-	if d.HasChange("tags") {
+	if d.HasChange("tags") && !attachedToPrivateEndpoint {
 		tagsRaw := d.Get("tags").(map[string]interface{})
 		payload.Tags = tags.Expand(tagsRaw)
 	}
 
-	err = client.CreateOrUpdateThenPoll(ctx, *id, *payload)
-	if err != nil {
-		return fmt.Errorf("updating %s: %+v", *id, err)
+	if propsOtherThanTagsUpdated || !attachedToPrivateEndpoint {
+		err = client.CreateOrUpdateThenPoll(ctx, *id, *payload)
+		if err != nil {
+			return fmt.Errorf("updating %s: %+v", *id, err)
+		}
+	}
+
+	if d.HasChange("tags") && attachedToPrivateEndpoint {
+		tagsRaw := d.Get("tags").(map[string]interface{})
+		tags := networkinterfaces.TagsObject{
+			Tags: tags.Expand(tagsRaw),
+		}
+		_, err = client.UpdateTags(ctx, *id, tags)
```

Comment:
Is it possible to just update Tags this way regardless of what else is being updated and whether it's a private NIC?

Reply:
@mbfrahry the problem with that approach is it unnecessarily increases exposure to the partial error problem (eg: when main body update is successful, but tags failed).

We're really a victim of poor API design here. I have submitted a request to my colleague on the NIC ARM API team to bake this logic into the PUT API, so we can be free of all these heavy lifting work.

Meanwhile we should keep this strategy of using a separate PATCH API call only when it's absolutely necessary.

Reply:
I see what you're saying. 

Do you have an idea of when the service team will be able to fix the issue on their end? If they can do it sooner rather than later, I'd say we don't take this PR in and just wait for the fix. If it's going to take a long time, please open an issue on the api-specs repo and link it in a comment in this pr so we can track it and remove this band-aid when it gets addressed

Reply:
@mbfrahry the fix won't be immediate. I have created the issue https://github.com/Azure/azure-rest-api-specs/issues/34437 on api-specs repo and propose we proceed with this change.

Reply:
@mbfrahry I have also added the issue link above as a comment in the code.